### PR TITLE
Simply aggregate exception happens in failed apphost deletion

### PR DESCRIPTION
**Pull request by wli3**
continue https://github.com/dotnet/core-setup/pull/7373

### CODE REVIEW

File: src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs [hunk 1]
```diff
@@ -88,20 +88,14 @@ public static class HostWriter
             }
             catch (Exception ex)
             {
-                FailedToDeleteApphostException failedToDeleteApphostException = null;
                 // Delete the destination file so we don't leave an unmodified apphost
                 try
                 {
                     File.Delete(appHostDestinationFilePath);
                 }
                 catch (Exception failedToDeleteEx) when (failedToDeleteEx is IOException || failedToDeleteEx is UnauthorizedAccessException)
```
Comment (nguerrera) [inline src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs]:
```suggestion
                catch (Exception failedToDeleteEx)
```

This way, we will not hide the original exception ever in the top-level msbuild error that says the task failed.

Comment (wli3) [reply src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs]: make sense

File: src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs [hunk 2]
```diff
@@ -88,20 +88,14 @@ public static class HostWriter
...
             catch (Exception ex)
             {
-                FailedToDeleteApphostException failedToDeleteApphostException = null;
                 // Delete the destination file so we don't leave an unmodified apphost
                 try
                 {
                     File.Delete(appHostDestinationFilePath);
                 }
-                catch (Exception failedToDeleteEx) when (failedToDeleteEx is IOException || failedToDeleteEx is UnauthorizedAccessException)
+                 catch (Exception failedToDeleteEx)
```
Comment (nguerrera) [inline src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs]: nit: Looks like I messed up the whitespace in my suggestion. Sorry.

Comment (wli3) [reply src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs]: fixing

Comment (wli3) [reply src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs]: done

**Comment by wli3** [comment]
@elinor-fung @jeffschwMSFT here is the small additional PR after  https://github.com/dotnet/core-setup/pull/7373

**Pull request merged**